MvvmCross/iOS/iOS/Views/Presenters/MvxIosViewPresenter.cs

```diff
@@ -111,48 +111,43 @@ public virtual void Show(IMvxIosView view, MvxViewModelRequest request)
             MvxRootPresentationAttribute attribute,
             MvxViewModelRequest request)
         {
+            // check if viewController is trying to initialize a navigation stack
+            if (attribute.WrapInNavigationController)
+            {
+                var navigationController = CreateNavigationController(viewController);
+                MasterNavigationController = navigationController as MvxNavigationController;
+                SetWindowRootViewController(navigationController);
+            }
+            else
+            // display the plain viewController as root
+            {
+                SetWindowRootViewController(viewController);
+
+                CloseMasterNavigationController();
+            }
+
             // check if viewController is a TabBarController
             if (viewController is IMvxTabBarViewController tabBarController)
             {
                 TabBarViewController = tabBarController;
-                SetWindowRootViewController(viewController);
 
-                CloseMasterNavigationController();
+                CloseTabBarViewController();
                 CleanupModalViewControllers();
-                CloseSplitViewController();
-
-                return;
             }
-
             // check if viewController is a SplitViewController
-            if (viewController is IMvxSplitViewController splitController)
+            else if (viewController is IMvxSplitViewController splitController)
             {
                 SplitViewController = splitController;
-                SetWindowRootViewController(viewController);
 
-                CloseMasterNavigationController();
+                CloseSplitViewController();
```

Comment:
Same as with the `TabBarViewController`, why first assign the `SplitViewController ` and after that directly destroy it by calling `CloseSplitViewController`